### PR TITLE
[Chore](agg-state) add sessionvariable enable_agg_state

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes

add session variable enable_agg_state

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java [hunk 1]
```diff
@@ -1229,7 +1229,7 @@ public String getNodeExplainString(String prefix, TExplainLevel detailLevel) {
     public int getNumInstances() {
         // In pipeline exec engine, the instance num equals be_num * parallel instance.
         // so here we need count distinct be_num to do the work. make sure get right instance
-        if (ConnectContext.get().getSessionVariable().enablePipelineEngine()) {
+        if (ConnectContext.get().getSessionVariable().enablePipelineEngine) {
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java:1232]: SessionVariable.enablePipelineEngine?

Comment (BiteTheDDDDt) [reply fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java:1232]: function is diffrent with public member

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 39.78 seconds
 stream load tsv:          459 seconds loaded 74807831229 Bytes, about 155 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          56 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          68.7 seconds inserted 10000000 Rows, about 145K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230703111217_clickbench_pr_171455.html

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 37.44 seconds
 stream load tsv:          451 seconds loaded 74807831229 Bytes, about 158 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          56 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          70.5 seconds inserted 10000000 Rows, about 141K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230704040038_clickbench_pr_171760.html

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**